articles/bottom-navigation-bar-in-android/index.md

```diff
@@ -0,0 +1,277 @@
+
+
+### Step 5 - Adding Details and Icons to the Bottom Navigation View
+First, let us add the icons required. In the `res/drawable` directory, right-click the `drawable` folder. Select `New → Vector Asset`.
+
+![Vector Asset](/engineering-education/bottom-navigation-bar-in-android/vector.jpg)
+
+* **_Asset type_** - provides an option of using a clipart or uploading a local file.
+* **_Name_** - name of the icon.
+* **_Size_**- the dimensions of the clipart
+* **_Color_** - provides color options for the clipart.
+* **_Opacity_** - deals with the visibility of the icon.
+
+Select the icon next to clipart. A list of different cliparts is displayed.
+
+![Clipart List](/engineering-education/bottom-navigation-bar-in-android/list.jpg)
+
+One can search for a clipart in the search bar and choose whether the design of the clipart should be filled or outlined. We will use the outlined design.
+
+![Favorites](/engineering-education/bottom-navigation-bar-in-android/favorites.jpg)
+
+Once a clipart is selected, click `OK → Next → Finish`. Open the menu file in `res/menu/bottom_navigation_menu` and add the following lines of code:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <item
+        android:id="@+id/navigation_home"
+        android:icon="@drawable/ic_outline_home_24"
+        android:title="@string/title_home" />
+    <item
+        android:id="@+id/navigation_search"
+        android:icon="@drawable/ic_outline_search_24"
+        android:title="@string/title_search" />
+    <item
+        android:id="@+id/navigation_favourites"
+        android:icon="@drawable/ic_baseline_favorite_border_24"
+        android:title="@string/title_favourites" />
+    <item
+        android:id="@+id/navigation_profile"
+        android:icon="@drawable/ic_baseline_perm_identity_24"
+        android:title="@string/title_profile" />
+
+</menu>
+
+```
+The **menu** file serves as a container for menu items. **Item** represents one item on the menu. Each item has a title and an icon.
+
+#### Step 6 - Let's Code.
+In our  `MainActivity.java` add the following lines of code:
+
+```java
+//Initialize Bottom Navigation View
+
+BottomNavigationView navView = findViewById(R.id.bottomNav_view);
+```
+We will initialize the Bottom Navigation view in the Activity's `onCreate` method. Next, we will initialize an `AppBarConfiguration` object. It is used to manage the behavior of the icons in the Navigation View.
+
+```java
+AppBarConfiguration appBarConfiguration = new AppBarConfiguration.Builder
+```
+We will pass the ID of the different destinations in the Bottom Navigation View.
+
+```java
+  R.id.navigation_home, R.id.navigation_favourites, R.id.navigation_profile, R.id.navigation_search )
```

Comment:
Something is missing here.
From your code on gihub, you can just merge the `AppBarConfiguration` with the navigation items and explain as a whole block

Reply:
@LinusMuema Solved it. Kindly check if it is ok.
Regards.